### PR TITLE
[US-011] Add add-update-delete options

### PR DESCRIPTION
**Pull request by viktor-debre**

### CODE REVIEW

File: Controllers/TaskController.cs [hunk 1]
```diff
@@ -58,5 +59,43 @@ public async Task<IActionResult> EndTask(int id)
...
+            ViewBag.AuthorizedUser = await _accountService.GetUserByEmail(User.Identity.Name);
+            return PartialView();
+        }
+        
+        public async Task<IActionResult> NewTask(NewTaskModel newTask)  
+        {
+            User currentUser = await _accountService.GetUserByEmail(User.Identity.Name);
+            await _taskService.AddTaskForUser(newTask, currentUser.Id);
+            return RedirectToAction("MyTasks", "Task");
+        }
```
Comment (KapNelson) [inline Controllers/TaskController.cs:75]: The controller should contain only methods that mapping some kind of request

File: Models/ViewModels/NewTaskModel.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    public class NewTaskModel
+    {
+        public int Id { get; set; }
+        [Required(ErrorMessage = "Not specified title")]
+        public string Title { get; set; }
+        public string? Description { get; set; }
+        [Required(ErrorMessage = "Not specified type of task")]
+        public string TypeOfTask { get; set; }
+    }
+}
```
Comment (KapNelson) [inline Models/ViewModels/NewTaskModel.cs:14]: Minor: perhaps we can somehow reuse an existing model?

File: Views/Task/Issues.cshtml [hunk 3]
```diff
@@ -1,7 +1,9 @@
+@using LawyerTimeTracker.ViewModels
 @model List<Issue>
 
 @{
     var issues = Model;
+    Issue updatingIssue = new Issue();
```
Comment (KapNelson) [inline Views/Task/Issues.cshtml:6]: Where are you using this variable?

**Review by KapNelson — CHANGES_REQUESTED** [review]

File: Models/ViewModels/NewTaskModel.cs (hunk 2 above)
Comment (viktor-debre) [reply Models/ViewModels/NewTaskModel.cs:14]: It's true but then we need to move validation to javascript, in the future, we can do that.

File: Views/Task/Issues.cshtml (hunk 3 above)
Comment (viktor-debre) [reply Views/Task/Issues.cshtml]: Yes, it is redundant, i removed it.

File: Controllers/TaskController.cs (hunk 1 above)
Comment (SofaDeveloper) [reply Controllers/TaskController.cs:76]: I think we can combine this logic into 1 method.

File: Controllers/TaskController.cs [hunk 4]
```diff
@@ -58,5 +59,43 @@ public async Task<IActionResult> EndTask(int id)
...
+        
+        [HttpPost]
+        public async Task<IActionResult> UpdateTask(NewTaskModel issue)
+        {
+            Issue currentIssue = await _taskService.GetTaskById(issue.Id);
+            currentIssue.Title = issue.Title;
+            currentIssue.Description = issue.Description;
+            currentIssue.TypeOfTask = issue.TypeOfTask;
+            await _taskService.UpdateTask(currentIssue);
```
Comment (SofaDeveloper) [inline Controllers/TaskController.cs:90]: This logic should be in service

**Review by SofaDeveloper — CHANGES_REQUESTED** [review]

File: Controllers/TaskController.cs [hunk 5]
```diff
@@ -58,5 +59,43 @@ public async Task<IActionResult> EndTask(int id)
...
+            return RedirectToAction("MyTasks", "Task");
+        }
```
Comment (SofaDeveloper) [inline Controllers/TaskController.cs:92]: This logic (86-90 lines) should be in service

**Review by SofaDeveloper — CHANGES_REQUESTED** [review]

**Review by SofaDeveloper — APPROVED** [review]

**Review by KapNelson — APPROVED** [review]

**Pull request merged**